api/test/wfl/tools/workflows.clj

```diff
@@ -42,8 +42,8 @@
    type object))
 
 (defn foldl
-  "Use the workflow `type` to left-fold the `object`, calling `f` on the values
-   with primitive types with the current state."
+  "Use ternary function `f` to reduce `object` tree of `type` starting
+  from `init` while preserving the types of intermediate reductions."
```

Comment:
@ehigham (-How wrong is this new docstring?-)

Reply:
While preserving the types of intermediate reductions? what does that mean?

Reply:
You could change `f` to take [result-so-far [type value]]` so it looks like "reduce" if that helps.

Reply:
The string tags are the "types".
Reductions are the intermediate result subtrees such as might be returned by `clojure.core/reductions`.
This differs from a generic left fold (or `reduce`) in that it treats one of the parameters differently.
Perhaps in your terminology: this is a left fold over a tree with an implicit lens.
(FWIW: I don't care where the grouping happens. This doesn't have to look like `reduce`.)

Reply:
Mostly, I just want some mention of the roles of the parameters to `foldl`,
and to cast it in Clojure terms so it is familiar to the team (or me, at least).
Please give me a thumbs up, a docstring more acceptable to you, or a "revert it".
Thanks.